lib/matplotlib/axes/_axes.py

```diff
@@ -55,7 +55,11 @@ def _plot_args_replacer(args, data):
         return ["y"]
     elif len(args) == 2:
         # this can be two cases: x,y or y,c
-        if not args[1] in data:
+        if (not args[1] in data and
+            not (hasattr(data, 'dtype') and
```

Comment:
This check is correct, but a bit hard to read. I would just try ... except. The cleanest way would be a little helper function:
~~~
def _has_name(data, name):
    try:
        return name in data or name in data.dtype.names
    except (AttributeError, TypeError):
        return False
~~~

and then
~~~
if _has_name(data, args[1]):
    return ["y", "c"]
~~~